### DISCUSSION TITLE
Connery SDK with Mistral AI

### DISCUSSION THREAD

**Discussion by RitaVerissimo86** [connery-io/connery-sdk#118] [Q&A] [answered]
Hi,

I'm interested in using Connery SDK but I'm using Mistral AI not Open AI.
Does the gmail tool and summary tool work in that case?

Thanks in advance!
Rita

**Comment by machulav** [comment]
Hi @RitaVerissimo86,

May I ask you to describe your use case in more details? So I can better understand and answer your question.

**Reply by RitaVerissimo86** [reply]
Hi @machulav ,
In the langchain documentation regarding Connery SDK: https://js.langchain.com/docs/integrations/tools/connery
The example mentioned to use the gmail plugin is with OpenAI:
```javascript
import { ConneryService } from "@langchain/community/tools/connery";
import { ChatOpenAI } from "@langchain/openai";
import { initializeAgentExecutorWithOptions } from "langchain/agents";

// Specify your Connery Runner credentials.
process.env.CONNERY_RUNNER_URL = "";
process.env.CONNERY_RUNNER_API_KEY = "";

// Specify OpenAI API key.
process.env.OPENAI_API_KEY = "";

// Specify your email address to receive the emails from examples below.
const recepientEmail = "test@example.com";

// Get the SendEmail action from the Connery Runner by ID.
const conneryService = new ConneryService();
const sendEmailAction = await conneryService.getAction(
  "CABC80BB79C15067CA983495324AE709"
);

// Run the action manually.
const manualRunResult = await sendEmailAction.invoke({
  recipient: recepientEmail,
  subject: "Test email",
  body: "This is a test email sent by Connery.",
});
console.log(manualRunResult);

// Run the action using the OpenAI Functions agent.
const llm = new ChatOpenAI({ temperature: 0 });
const agent = await initializeAgentExecutorWithOptions([sendEmailAction], llm, {
  agentType: "openai-functions",
  verbose: true,
});
const agentRunResult = await agent.invoke({
  input: `Send an email to the ${recepientEmail} and say that I will be late for the meeting.`,
});
console.log(agentRunResult);
```
My question is: can this code be adapted to work with Mistral AI instead of OpenAI?

Thanks ;)
R.

**Reply by machulav** [reply, marked as answer]
Thanks for the clarification!

Because Connery is fully integrated with LangChain, every Connery action is wrapped into the native LangChain tool. This means you can use the Connery action freely in the LangChain ecosystem, including all the models LangChain supports.

So yes, you can use Connery actions with Mistral AI. Here's an example of how to call tools using Mistral AI and LangChain: https://js.langchain.com/docs/integrations/chat/mistral#tool-calling. Just use the Connery tools/toolkit in this example, and it should work.

Important note: Connery SDK v0.2.0 and later do not work with LangChain yet, but we're working on it right now. In the meantime, you can use Connery [v0.0.8](https://github.com/connery-io/connery/releases/tag/v0.0.8) (and the related CLI: connery@0.0.40), which is supported by LangChain.

Please let me know if you have any questions.

**Reply by RitaVerissimo86** [reply]
Thanks @machulav 👍 🥇
